tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,599 @@
+
+  bool isNetworkOutput() const { return false; }
+
+  void setBroadcastAcrossBatch(bool broadcastAcrossBatch) {}
+
+  bool getBroadcastAcrossBatch() const { return false; }
+
+  nvinfer1::TensorLocation getLocation() const { return location_; }
+
+  void setLocation(nvinfer1::TensorLocation location) {
+    location_ = location;
+  }
+  bool setDynamicRange(float min, float max) {
+    dynamic_range_ = std::max(std::abs(min), std::abs(max));
+    return true;
+  }
+  
+  float getDynamicRange() const { return dynamic_range_; }
+  bool dynamicRangeIsSet() const { return true; }
+
+  void resetDynamicRange() {}
+
+  float getDynamicRangeMin() const { return 0.f; }
+
+  float getDynamicRangeMax() const { return 0.f; }
+
+  void setAllowedFormats(nvinfer1::TensorFormats formats) {}
+
+  nvinfer1::TensorFormats getAllowedFormats() const { return 1; }
+
+  bool isShapeTensor() const { return false; }
+  bool isExecutionTensor() const { return true; }
+
+ private:
+  nvinfer1::DataType trt_dtype_;
+  nvinfer1::Dims trt_dims_;
+  std::string name_;
+  nvinfer1::TensorLocation location_;
+  float dynamic_range_;
+};
+
+// Fake ITensor implementation for testing purposes.
+class FakeITensor {
+ public:
+  FakeITensor() : dynamic_range_(0.0f) {}
+
+  FakeITensor(const nvinfer1::Dims& dims) : dims_(dims), dynamic_range_(0.0f) {}
+
+  FakeITensor(const std::vector<int>& dims)
+      : dims_(GetTestDims(dims)), dynamic_range_(0.0f) {}
+
+  void setName(const char* name) { name_ = name; }
+
+  const char* getName() const { return name_.c_str(); }
+
+  void setDimensions(nvinfer1::Dims dimensions) { dims_ = dimensions; }
+
+  nvinfer1::Dims getDimensions() const { return dims_; }
+
+  void setType(nvinfer1::DataType type) { type_ = type; }
+
+  nvinfer1::DataType getType() const { return type_; }
+
+  bool isNetworkInput() const { return false; }
+
+  bool isNetworkOutput() const { return false; }
+
+  void setBroadcastAcrossBatch(bool broadcastAcrossBatch) {}
+
+  bool getBroadcastAcrossBatch() const { return false; }
+
+  nvinfer1::TensorLocation getLocation() const { return location_; }
+
+  void setLocation(nvinfer1::TensorLocation location) {
+    location_ = location;
+  }
+
+  bool setDynamicRange(float min, float max) {
+    dynamic_range_ = std::max(std::abs(min), std::abs(max));
+    return true;
+  }
+
+  bool dynamicRangeIsSet() const { return true; }
+
+  void resetDynamicRange() {}
+
+  float getDynamicRangeMin() const { return 0.f; }
+
+  float getDynamicRangeMax() const { return 0.f; }
+
+  void setAllowedFormats(nvinfer1::TensorFormats formats) {}
+
+  nvinfer1::TensorFormats getAllowedFormats() const { return 1; }
+
+  bool isShapeTensor() const { return false; }
+  bool isExecutionTensor() const { return true; }
+
+ private:
+  std::string name_;
+  nvinfer1::Dims dims_;
+  nvinfer1::DataType type_;
+  nvinfer1::TensorLocation location_;
+  float dynamic_range_;
+};
+
+enum class TensorType : int
+{
+  kNONE,
```

Comment:
We shouldn't need kNONE, right?

Reply:
Shifting over to an exhaustive switch as you mention below sounds like an improvement, resolving this conversation and using that comment for the changes.